bedwars-plugin/src/main/java/com/andrei1058/bedwars/arena/Arena.java

```diff
@@ -2552,13 +2559,24 @@ public boolean startReSpawnSession(Player player, int seconds) {
                     PaperSupport.teleportC(player, getReSpawnLocation(), PlayerTeleportEvent.TeleportCause.PLUGIN);
                 }, 10L);
             } else {
+                player.setHealth(20);
+                player.getActivePotionEffects().forEach(effect -> player.removePotionEffect(effect.getType()));
+                player.setFireTicks(0);
                 ITeam team = getTeam(player);
                 team.respawnMember(player);
             }
             return true;
         }
         return false;
     }
+    
+    public static void checkPlayerHealth(Player p, EntityDamageEvent e, IArena a) {
+        if (e.getDamage() > p.getHealth() && e.getCause() != EntityDamageEvent.DamageCause.ENTITY_EXPLOSION) {
+            e.setCancelled(true);
+            Bukkit.getServer().getPluginManager().callEvent(new PlayerDeathEvent(p, Arrays.asList(p.getInventory().getContents()), p.getTotalExperience(), null));
+            Bukkit.getServer().getPluginManager().callEvent(new PlayerRespawnEvent(p, a.getReSpawnLocation(), false));
+        }
```

Comment:
why are you checking for DamageCause Explosion?

Reply:
if i do not then the player would instantly die when trying to fireball jump, there might be a better way to handle that

Reply:
The fireball jump thing is configurable, but default config should have some changes to be more user friendly. (no 1 click deaths haha)
Maybe that would solve the problem aswell as looking at the current config it might actually be configured that way. 
Not saying that it is wanted behaviour.